modules/ingester/instance.go

```diff
@@ -318,11 +318,11 @@ func (i *instance) tracesToCut(cutoff time.Duration, immediate bool) []*trace {
 	i.tracesMtx.Lock()
 	defer i.tracesMtx.Unlock()
 
-	now := time.Now()
+	cutoffTime := time.Now().Add(cutoff)
 	tracesToCut := make([]*trace, 0, len(i.traces))
 
 	for key, trace := range i.traces {
-		if now.Add(cutoff).After(trace.lastAppend) || immediate {
+		if cutoffTime.After(trace.lastAppend) || immediate {
```

Comment:
if you move immediate to the front, it will short circuit the cutoffTime calculation. this may reduce the amount of time tracesMtx is held

Reply:
immediate is almost never true.  it only occurs when the ingester is shutting down to force flush traces from the active trace map to disk.  i think this ordering is fine.

Reply:
fair enough, thanks for considering it.